charms/pipelines-persistence/reactive/pipelines_persistence.py

```diff
@@ -15,13 +15,15 @@ def update_image():
     clear_flag('charm.pipelines-persistence.started')
 
 
-@when('layer.docker-resource.oci-image.available')
+@when('layer.docker-resource.oci-image.available', 'pipelines-api.available')
 @when_not('charm.pipelines-persistence.started')
-def start_charm():
+def start_charm(api):
```

Comment:
I see this pattern being used in a few kubeflow charms.  Please check the dangers of endpoints-as-arguments:

https://charmsreactive.readthedocs.io/en/latest/charms.reactive.decorators.html#charms-reactive-decorators

Specifically how no-one ever remembers the bottom-up / left-to-right passage.  It's also easy to change a `when` to a `when_any` and lose the args.